webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/slack/BranchNotificationMessageBuilderSlack.java

```diff
@@ -132,4 +135,15 @@ String getSummaryString(List<String> strings) {
         .map(s -> StringUtils.abbreviate(s, STRING_IN_SUMMARY_ABRREVIATE_LENGHT))
         .collect(Collectors.joining(", "));
   }
+
+  protected String getTranslationsReadyMsg(String branchName) {
+    GithubBranchDetails branchDetails = new GithubBranchDetails(branchName);
```

Comment:
That notifier is not Github specific, so using `GithubBranchDetails` would be a problem when it is used for example with a change in Phabricator.

Reply:
let's make it an option to use or not `GithubBranchDetails` to process the branchName. The  config has to be added to `SlackConfigurationProperties` and then passed down as other current options

Reply:
@hylstonnb something like that, need to finish passing the option around but that's the idea

``` 
protected String getTranslationsReadyMsg(String branchName) {
    final Builder<String, Object> messageParamMap = ImmutableMap.<String, Object>builder();
    messageParamMap.put("branchName", branchName)
    if (isGithubPR) {
      GithubBranchDetails branchDetails = new GithubBranchDetails(branchName);
      messageParamMap.put("repository", branchDetails.getRepository());
    }
    MessageFormat messageFormat = new MessageFormat(translationsReadyMsg);
    return messageFormat.format(messageParamMap.build());
  }
```

Need to specify a new config, 
```
static class SlackConfigurationProperties {

    String slackClientId;

    MessageBuilderConfigurationProperties messages = new MessageBuilderConfigurationProperties();

    String userEmailPattern;

    boolean useDirectMessage = false;
    
    boolean isGithubPR = false;
```

Reply:
makes sense, addressed, thank you